mixnet/src/utils.rs

```diff
@@ -0,0 +1,125 @@
+use std::time::Duration;
+
+use rand::{Rng, SeedableRng};
+use rand_distr::{Distribution, Exp};
+
+/// A struct which helps give us a random interval from an exponential distribution.
+#[derive(Copy, Clone, Debug)]
+pub struct PoissonExpInterval<R>(R);
+
+impl<R: SeedableRng> PoissonExpInterval<R> {
+    /// Returns a new instance
+    pub fn new(seed: R::Seed) -> Self {
+        Self(R::from_seed(seed))
+    }
+}
+
+impl<R: Rng> PoissonExpInterval<R> {
+    /// Get a random interval from an exponential distribution in seconds.
+    ///
+    /// The Poisson (exponential) distribution is used because of its memoryless properties,
+    /// which is known to offer optimal anonymity properties (Nym whitepaper 4.4).
+    pub fn interval(&mut self, rate_per_min: f64) -> Duration {
```

Comment:
> Also, it would be helpful to add a comment in the first method, that why we use exponential distribution for intervals, even though the function name says 'poisson'. It's explained in the spec as a comment.

What I meant in https://github.com/logos-co/nomos-node/pull/575#discussion_r1472388701 is that we need to mention why we use "exponential" distribution even though the name of this struct starts with `Poisson`. Someone may don't understand it if there is no clear explanation. It would be enough to just copy and paste the explanation in [nomos-specs](https://github.com/logos-co/nomos-specs/blob/9a54d90d1411de99647373914450bcc57520b92b/mixnet/poisson.py#L5)

Reply:
I added comments: e6cdc9fb673cb73e87e949c65557efd65a1fb9a4. Also, I renamed the `PoissonExpInterval` struct to `Poisson`, because the struct has the `interval` function and we don't need to expose its implementation detail (the fact of using exp distribution for intervals) by its name.